get_point_grid.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import requests
+import numpy as np
+from typing import List, Tuple, Dict, Any
+import creds
+
+MAX_POINTS = 10000
+MIN_POINTS = 2
+
+def generate_grid(lat_start: float, lat_end: float, lon_start: float, lon_end: float, step: float) -> List[Tuple[float, float]]:
+    """Generate a grid of latitude and longitude coordinates, correctly handling both positive and negative steps."""
+    # Determine the direction of the step for latitude and longitude
+    lat_step = step if lat_end > lat_start else -step
+    lon_step = step if lon_end > lon_start else -step
+    
+    # Generate points using adjusted steps
+    lat_points = np.arange(lat_start, lat_end, lat_step)
+    lon_points = np.arange(lon_start, lon_end, lon_step)
+    
+    # Create grid using list comprehension
+    return [(lat, lon) for lat in lat_points for lon in lon_points]
+
+def snap_to_roads(coordinates: List[Tuple[float, float]]) -> List[Tuple[float, float]]:
+    """Snap a list of coordinates to the nearest roads using Google Roads API and return a list of lat/lon pairs."""
+    base_url = "https://roads.googleapis.com/v1/snapToRoads"
+    # Create the path parameter string from the coordinates list
+    path_param = "|".join([f"{lat},{lon}" for lat, lon in coordinates])
+    params = {"path": path_param, "interpolate": False, "key": creds.GOOGLE_API_KEY}
+    response = requests.get(base_url, params=params)
+    
+    if response.status_code == 200:
+        snapped_points = response.json().get('snappedPoints', [])
+        # Extract the latitude and longitude from each snapped point
+        return [(point['location']['latitude'], point['location']['longitude']) for point in snapped_points]
+    else:
+        print("API Request Failed:", response.text)
+        return []
+
+def parse_args() -> argparse.ArgumentParser:
+    """Parse command line arguments for the grid coordinates and API key, including grid resolution and maximum number of points."""
+    parser = argparse.ArgumentParser(description="Generate a grid of coordinates and snap them to the nearest roads.")
+    parser.add_argument("lat_start", type=float, help="Start latitude of the grid.")
```

Comment:
Curious if you have seen [click](https://click.palletsprojects.com/en/8.1.x/). I like it more now after seeing it at work.

Reply:
I haven't! Pretty cool! Feels like over engineering for this use case but will definitely consider it in the future